components/Chips/src/MDCChipField.h

```diff
@@ -0,0 +1,242 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "MaterialChips.h"
+#import "MaterialTextFields.h"
+
+/**
+ Note: There is a UIKit bug affecting iOS 8.0-8.2 where UITextFields do not properly call the
+ public |deleteBackward| method when the user presses backspace, which means that users can't
+ delete chips with the backspace key. Apple fixed the underlying bug in iOS 8.3, so clients
+ targeting a minimum iOS version of >= 8.3 are not affected by this bug.
+
+ As a workaround, this class can OPTIONALLY USE THE PRIVATE |keyboardInputShouldDelete| method to
+ detect when backspace is pressed and forward the event to |deleteBackward|. There is anecdotal
+ evidence that this private usage is currently acceptable by Apple, see
+ http://stackoverflow.com/a/25862878/23624.
+
+ To enable the workaround, define MDC_CHIPFIELD_PRIVATE_API_BUG_FIX to be 1 when building
+ MDCChipField. MDC itself will never define this macro, it's entirely a client decision.
+
+ If you target iOS versions < 8.3 and you decide not to define MDC_CHIPFIELD_PRIVATE_API_BUG_FIX,
+ your iOS 8.0-8.2 users will *not be able to delete chips using the backspace key*. Please
+ consider adding a delete button to each chip's accessoryView.
+
+ An alternative approach (using an empty character to detect backspace) makes MDCChipField unusable
+ by VoiceOver users on all versions of iOS.
+ */
+
+/** The minimum text field width of the text field contained within the MDCChipField. */
+extern const CGFloat kMDCChipFieldDefaultMinTextFieldWidth;
+
+/** The default content edge insets for the MDCChipField. */
+extern const UIEdgeInsets kMDCChipFieldDefaultContentEdgeInsets;
+
+@protocol MDCChipFieldDelegate;
+
+/** Defines when text field input is changed into a chip. */
+typedef NS_OPTIONS(NSUInteger, MDCChipFieldDelimiter) {
+  MDCChipFieldDelimiterNone = 0,
+  MDCChipFieldDelimiterReturn = 1 << 0,
+  MDCChipFieldDelimiterSpace = 1 << 1,
+  MDCChipFieldDelimiterDidEndEditing = 1 << 2,
+  MDCChipFieldDelimiterDefault = (MDCChipFieldDelimiterReturn | MDCChipFieldDelimiterDidEndEditing),
+  MDCChipFieldDelimiterAll = 0xFFFFFFFF
```

Comment:
When setting the delimiter to MDCChipFieldDelimiterAll, return stops working for some reason (tested example in simulator).

Reply:
Should be working now.